katello/katello-change-hostname

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/ruby
+
+require "socket"
+require 'highline/import'
+
+STDOUT.puts "Starting the hostname change..."
+
+STDOUT.puts("\nA hostname change requires a manual change in Satellite. \n
+            Please log into UI and rename the default capsule (under infrastructure->capsules) 
+            to the new hostname using both the name and url fields.")
+
+unless HighLine.agree('Have you made this necessary change and want to proceed with the hostname change? [yes/no]')
+  STDOUT.puts "Hostname change aborted, no changes have been made to your system"
+  exit
+end
+
+if ARGV[0] && ARGV.count == 1
+  NEW_HOSTNAME = ARGV[0]
+else
+  STDOUT.puts "USAGE: katello-change-hostname <new-hostname>"
+  exit
+end
+
+# Get the hostname from your system
+OLD_HOSTNAME = Socket.gethostname
+
+RHEL_VERSION = `rpm -q --queryformat '%{VERSION}' redhat-release-server`[0]
+
+STDOUT.puts "stopping services"
+`katello-service stop`
+
+STDOUT.puts "deleting old certs"
+`
+rm -rfv /etc/pki/katello{,.bak}
+rm -rfv /etc/pki/katello-certs-tools{,.bak}
+rm -rfv /etc/candlepin/certs/amqp{,.bak}
+rm -rfv /etc/foreman-proxy/*ssl*
+rm -rfv /etc/foreman/old-certs
+rm -rfv /etc/foreman/*.pem
+rm -rfv /var/lib/puppet/ssl
+rm -rfv /root/ssl-build
+`
+
+if RHEL_VERSION.to_i == 6
+  STDOUT.puts "updating hostname in /etc/sysconfig/network"
+  `sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/sysconfig/network`
+elsif RHEL_VERSION.to_i == 7
+  STDOUT.puts "updating hostname in /etc/hostname"
+  `sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hostname`
+
+  STDOUT.puts "setting hostname"
+  `hostnamectl set-hostname #{NEW_HOSTNAME}`
+else
+  STDOUT.puts "couldn't find RHEL version: #{RHEL_VERSION}"
+  exit
+end
+
+STDOUT.puts "updating hostname in /etc/hosts"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/hosts`
+
+STDOUT.puts "checking if hostname was changed"
+hostname = Socket.gethostname
+if hostname != NEW_HOSTNAME
+  STDOUT.puts "the new hostname was not changed, check hostname -f"
+  exit
+end
+
+STDOUT.puts "updating hostname in /etc/foreman-installer/scenarios.d/satellite-answers.yaml"
+`sed -i -e 's/#{OLD_HOSTNAME}/#{NEW_HOSTNAME}/g' /etc/foreman-installer/scenarios.d/satellite-answers.yaml`
```

Comment:
I would think it could, you would have to know each option that needs changing which might be burdensome. However, a hook in the installer would be able to load the answers in and make this change I'd think.